### PR TITLE
Add the SLBeta6 Release Note

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add the SLBeta6 release note.
> Fixes #

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

**Review by niveathika — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Added the `--with-tests` flag for the OpenAPI client generation command to generate boiler-plate test files of the relevant remote functions. For example,
+
+  > `bal openapi -i <openapi contract> --mode client --with-test`
+
+#### Ballerina to OpenAPI Command Improvements
+
+Added the `--export-openapi` flag for the `bal build` command to generate OpenAPI contract files for all the services in the current package. For example,
+
+  > `bal build --export-openapi`
```
Comment (NipunaRanasinghe) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:101]: Shouldn't both of these improvements be listed under `#### OpenAPI Tool` header? We don't use L4 headers for the improvement items in the release note, I guess. @praneesha @lnash94 WDYT?

Comment (lnash94) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:101]: yeah, it should be, sorry for the mistake, I will change.

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+#### `http` Package
+- Changed the `RequestContext:add` function to `RequestContext:set`.
+- Allowed listener-level interceptors to have only the default path.
+- Improved the `parseHeader()` function to support multiple header values.
+
+## Developer Tools Updates
+
+### Improvements
+
```
Comment (lnash94) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:90]:
```suggestion

#### OpenAPI Tool

```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#### OpenAPI to Ballerina Command Improvements
```
Comment (lnash94) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:91]:
```suggestion
##### OpenAPI to Ballerina Command Improvements
```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+Added the `--with-tests` flag for the OpenAPI client generation command to generate boiler-plate test files of the relevant remote functions. For example,
+
+  > `bal openapi -i <openapi contract> --mode client --with-test`
+
+#### Ballerina to OpenAPI Command Improvements
```
Comment (lnash94) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:97]:
```suggestion
##### Ballerina to OpenAPI Command Improvements
```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+## Standard Library Updates
+
+### New Features
+
+#### `graphql` Package
+Added support for the GraphQL list type inputs.
+
+#### `http` Package
+Introduced request and request error interceptors at the service level.
+
```
Comment (daneshk) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:79]:
```suggestion

#### `xmldata` Package
Added support to convert an XML to a Record.
```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+### Improvements
```
Comment (BuddhiWathsala) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:80]:
```suggestion
#### `time` Package
Add time zone handling APIs.


### Improvements
```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+## Developer Tools Updates
+
+### Improvements
+
+#### OpenAPI to Ballerina Command Improvements
+
+Added the `--with-tests` flag for the OpenAPI client generation command to generate boiler-plate test files of the relevant remote functions. For example,
+
+  > `bal openapi -i <openapi contract> --mode client --with-test`
```
Comment (lnash94) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:95]:
```suggestion
  > `bal openapi -i <openapi contract> --mode client --with-tests`
```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md (hunk 1 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md]: Done.

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+Added support for the GraphQL list type inputs.
+
+#### `http` Package
+Introduced request and request error interceptors at the service level.
+
+#### `time` Package
+Added time zone handling support.
+
+#### `time` Package
+Add time zone handling APIs.
```
Comment (daneshk) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:84]:
```suggestion
```

Comment (daneshk) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:84]: Shall we remove one? seems like duplicated points

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+Added the `--with-tests` flag for the OpenAPI client generation command to generate boiler-plate tests file of the relevant remote functions. For example,
+
+  > `bal openapi -i <openapi contract> --mode client --with-tests`
+
+##### Ballerina to OpenAPI Command Improvements
+
+Added the `--export-openapi` flag for the `bal build` command to generate OpenAPI contract files for all the services in the current package. For example,
+
+  > `bal build --export-openapi`
```
Comment (NipunaRanasinghe) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:110]:
```suggestion
#### OpenAPI Tool

- Added `--with-tests` option in the OpenAPI client generation command, which will generate test file templates for all the relevant remote functions of the client. For example,

  > `bal openapi -i <openapi contract> --mode client --with-tests`

- Added `--export-openapi` option in the `build` command, which will generate OpenAPI contract files for all the services in the current package. For example,

  > `bal build --export-openapi`
```

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md]: Need to add "the"article as the referred option is a noun for grammatical accuracy.

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md]: Done.

**Review by chamil321 — APPROVED** [review]

**Review by lnash94 — APPROVED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]

**Comment by azinneera** [comment]
Breaking changes in Test Framework LGTM. But the improvements to CLI and Ballerina.toml changes are missing.

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+## Standard Library Updates
+
+### New Features
+
+#### `graphql` Package
+Added support for the GraphQL list type inputs.
+
+#### `http` Package
+Introduced request and request error interceptors at the service level.
+
```
Comment (daneshk) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:78]:
```suggestion

#### `xmldata` Package
Added support to convert an XML to a Record

```

**Review by daneshk — APPROVED** [review]

**Comment by praneesha** [comment]
> Breaking changes in Test Framework LGTM. But the improvements to CLI and Ballerina.toml changes are missing.

Added.

**Review by azinneera — APPROVED** [review]
Ballerina packages and Test Framework release note LGTM

**Review by BuddhiWathsala — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+### Bug Fixes
+
+To view bug fixes, see the GitHub milestone for Swan Lake Beta3 of the repositories below.
+
+- [Language Server](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta6%22+label%3AType%2FBug+label%3ATeam%2FLanguageServer)
+- [OpenAPI](https://github.com/ballerina-platform/ballerina-openapi/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta6%22+label%3AType%2FBug)
+- [Debugger](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3AType%2FBug+label%3AArea%2FDebugger+milestone%3A%22Ballerina+Swan+Lake+-+Beta6%22+is%3Aclosed)
+
+### Ballerina Packages
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:142]: It seems the level of this heading is incorrect.

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+
+### Bug Fixes
+
+To view bug fixes, see the GitHub milestone for Swan Lake Beta3 of the repositories below.
+
+- [Language Server](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta6%22+label%3AType%2FBug+label%3ATeam%2FLanguageServer)
+- [OpenAPI](https://github.com/ballerina-platform/ballerina-openapi/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%22Ballerina+Swan+Lake+-+Beta6%22+label%3AType%2FBug)
+- [Debugger](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+label%3AType%2FBug+label%3AArea%2FDebugger+milestone%3A%22Ballerina+Swan+Lake+-+Beta6%22+is%3Aclosed)
+
+## Ballerina Packages
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:142]:
```suggestion
## Ballerina Packages Updates
```

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+name = "googleapis.gmail"
+version = "0.1.0”
+icon = “images/icon.png”
+```
+
+>**Info:** The specified icon file will be packaged into the `docs/` directory of the Ballerina archive (`.bala`). Only the `.png` format is supported for the icon.
+
+### Breaking Changes
+
+#### Test Framework
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:159]: Why `Test Framework` is listed under `Ballerina Packages`? Shouldn't we have it under `Developer Tools Updates` @azinneera

**Review by anupama-pathirage — APPROVED** [review]

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md (hunk 12 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md (hunk 13 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-beta6/RELEASE_NOTE.md:136]: Fixed.

**Pull request merged**